SConstruct

```diff
@@ -731,9 +731,12 @@ if env.msvc:
         env.Append(CCFLAGS=["/Od"])
 else:
     if env["debug_symbols"]:
-        # Adding dwarf-4 explicitly makes stacktraces work with clang builds,
-        # otherwise addr2line doesn't understand them
-        env.Append(CCFLAGS=["-gdwarf-4"])
+        if env["platform"] == "windows":
+            env.Append(CCFLAGS=["-gdwarf-5"])
+        else:
+            # Adding dwarf-4 explicitly makes stacktraces work with clang builds,
+            # otherwise addr2line doesn't understand them
+            env.Append(CCFLAGS=["-gdwarf-4"])
```

Comment:
For the record, this was added in #71437 by @vaartis, but I researched it again to understand better in which case it's actually needed. We might want to reconsider this override.

It seems like dwarf-5 is the new default since GCC 11, but it depends on GNU binutils version 2.35 for `addr2line` to support it.

https://gcc.gnu.org/gcc-11/changes.html
https://fedoraproject.org/wiki/Changes/BINUTILS235

From a quick look at https://pkgs.org/download/binutils, the only supported Linux distro that doesn't have at least binutils 2.35 is Ubuntu 20.04, which is still on 2.34.

Maybe losing support for backtraces on Ubuntu 20.04 is a reasonable compromise now to have the latest dwarf format, which according to GCC release notes "can produce up to 25% more compact debug information compared to earlier versions."

Now, this was originally reported on #devel by @adamscott in Jan 2023 and I doubt that he was running Ubuntu 20.04 at the time, so that's weird.

We should also evaluate whether macOS could use `dwarf-5` too.

Reply:
I tested on Fedora 41 with clang 19.1.4 and binutils 2.43.1, compiling with `scons platform=linuxbsd target=editor dev_build=yes dev_mode=yes debug_symbols=yes use_llvm=yes linker=mold optimize=none fast_unsafe=yes` (same command as @adamscott on #devel in Jan 2023), and I get proper stacktraces:
```
================================================================
handle_crash: Program crashed with signal 11
Engine version: Godot Engine v4.4.dev.custom_build (aa8d9b83f66488dbb2c9c918e9016ef80f821fb4)
Dumping the backtrace. Please include this when reporting the bug to the project developer.
[1] /lib64/libc.so.6(+0x19dd0) [0x7f613d2fadd0] (??:0)
[2] RVO2D::KdTree2D::queryAgentTreeRecursive(RVO2D::Agent2D*, float&, unsigned long) const (/home/akien/Godot/godot/thirdparty/rvo2/rvo2_2d/KdTree2d.cpp:257)
[3] RVO2D::KdTree2D::computeAgentNeighbors(RVO2D::Agent2D*, float&) const (/home/akien/Godot/godot/thirdparty/rvo2/rvo2_2d/KdTree2d.cpp:239)
[4] RVO2D::Agent2D::computeNeighbors(RVO2D::RVOSimulator2D*) (/home/akien/Godot/godot/thirdparty/rvo2/rvo2_2d/Agent2d.cpp:53)
[5] NavMap::compute_single_avoidance_step_2d(unsigned int, NavAgent**) (/home/akien/Godot/godot/modules/navigation/nav_map.cpp:771)
[6] WorkerThreadPool::GroupUserData<NavMap, void (NavMap::*)(unsigned int, NavAgent**), NavAgent**>::callback_indexed(unsigned int) (/home/akien/Godot/godot/./core/object/worker_thread_pool.h:209)
[7] WorkerThreadPool::_process_task(WorkerThreadPool::Task*) (/home/akien/Godot/godot/core/object/worker_thread_pool.cpp:102)
[8] WorkerThreadPool::_thread_function(void*) (/home/akien/Godot/godot/core/object/worker_thread_pool.cpp:184)
[9] Thread::callback(unsigned long, Thread::Settings const&, void (*)(void*), void*) (/home/akien/Godot/godot/core/os/thread.cpp:66)
[10] void std::__invoke_impl<void, void (*)(unsigned long, Thread::Settings const&, void (*)(void*), void*), unsigned long, Thread::Settings, void (*)(void*), void*>(std::__invoke_other, void (*&&)(unsigned long, Thread::Settings const&, void (*)(void*), void*), unsigned long&&, Thread::Settings&&, void (*&&)(void*), void*&&) (/usr/bin/../lib/gcc/x86_64-redhat-linux/14/../../../../include/c++/14/bits/invoke.h:61)
[11] std::__invoke_result<void (*)(unsigned long, Thread::Settings const&, void (*)(void*), void*), unsigned long, Thread::Settings, void (*)(void*), void*>::type std::__invoke<void (*)(unsigned long, Thread::Settings const&, void (*)(void*), void*), unsigned long, Thread::Settings, void (*)(void*), void*>(void (*&&)(unsigned long, Thread::Settings const&, void (*)(void*), void*), unsigned long&&, Thread::Settings&&, void (*&&)(void*), void*&&) (/usr/bin/../lib/gcc/x86_64-redhat-linux/14/../../../../include/c++/14/bits/invoke.h:96)
[12] void std::thread::_Invoker<std::tuple<void (*)(unsigned long, Thread::Settings const&, void (*)(void*), void*), unsigned long, Thread::Settings, void (*)(void*), void*> >::_M_invoke<0ul, 1ul, 2ul, 3ul, 4ul>(std::_Index_tuple<0ul, 1ul, 2ul, 3ul, 4ul>) (/usr/bin/../lib/gcc/x86_64-redhat-linux/14/../../../../include/c++/14/bits/std_thread.h:301)
[13] std::thread::_Invoker<std::tuple<void (*)(unsigned long, Thread::Settings const&, void (*)(void*), void*), unsigned long, Thread::Settings, void (*)(void*), void*> >::operator()() (/usr/bin/../lib/gcc/x86_64-redhat-linux/14/../../../../include/c++/14/bits/std_thread.h:308)
[14] std::thread::_State_impl<std::thread::_Invoker<std::tuple<void (*)(unsigned long, Thread::Settings const&, void (*)(void*), void*), unsigned long, Thread::Settings, void (*)(void*), void*> > >::_M_run() (/usr/bin/../lib/gcc/x86_64-redhat-linux/14/../../../../include/c++/14/bits/std_thread.h:253)
[15] /home/akien/Godot/godot/bin/godot.linuxbsd.editor.dev.x86_64.llvm() [0xb46a0b4] (msdf-error-correction.cpp:?)
[16] /lib64/libc.so.6(+0x70cd7) [0x7f613d351cd7] (??:0)
[17] /lib64/libc.so.6(+0xf4c8c) [0x7f613d3d5c8c] (??:0)
-- END OF BACKTRACE --
================================================================
```

This would need testing on older distros to be sure.

Reply:
> Now, this was originally reported on #devel by @adamscott in Jan 2023 and I doubt that he was running Ubuntu 20.04 at the time, so that's weird.

I was running contemporary Fedora.